src/mscorlib/shared/System/SpanHelpers.Char.cs

```diff
@@ -167,6 +168,91 @@ public static unsafe int IndexOf(ref char searchSpace, char value, int length)
             }
         }
 
+        public static unsafe int LastIndexOf(ref char firstChar, int startIndex, char value, int length)
+        {
+            Debug.Assert(length >= 0);
+            
+            fixed (char* pChars = &firstChar)
+            {
+                char* pCh = pChars + startIndex;
+                char* pEndCh = pCh - length;
+
+#if !netstandard11
+                if (Vector.IsHardwareAccelerated && length >= Vector<ushort>.Count * 2)
+                {
+                    const int elementsPerByte = sizeof(ushort) / sizeof(byte);
+                    length = (((int)pCh & (Unsafe.SizeOf<Vector<ushort>>() - 1)) / elementsPerByte) + 1;
+                }
+            SequentialScan:
+#endif
+                while (length >= 4)
+                {
+                    length -= 4;
+
+                    if (*pCh == value)
+                        goto Found;
+                    if (*(pCh - 1) == value)
+                        goto Found1;
+                    if (*(pCh - 2) == value)
+                        goto Found2;
+                    if (*(pCh - 3) == value)
+                        goto Found3;
+
+                    pCh -= 4;
+                }
+
+                while (length > 0)
+                {
+                    length -= 1;
+
+                    if (*pCh == value)
+                        goto Found;
+
+                    pCh -= 1;
+                }
+#if !netstandard11
+                // We get past SequentialScan only if IsHardwareAccelerated is true. However, we still have the redundant check to allow
+                // the JIT to see that the code is unreachable and eliminate it when the platform does not have hardware accelerated.
+                if (Vector.IsHardwareAccelerated && pCh > pEndCh)
+                {
+                    length = (int)((pCh - pEndCh) & ~(Vector<ushort>.Count - 1));
+
+                    // Get comparison Vector
+                    Vector<ushort> vComparison = new Vector<ushort>(value);
+
+                    while (length > 0)
+                    {
+                        char* pStart = pCh - Vector<ushort>.Count + 1;
+                        Vector<ushort> vMatches = Vector.Equals(vComparison, Unsafe.ReadUnaligned<Vector<ushort>>(pStart));
```

Comment:
We use `Unsafe.Read` in IndexOf, but `Unsafe.ReadUnaligned` here.

Reply:
I am not sure if `pStart` is guaranteed to be vector aligned here (unlike the case of IndexOf). I will have to reason about it to convince myself. If it is aligned, I will change it to Unsafe.Read.

Reply:
It should be guaranteed, or else the preamble that we are doing is worthless.  The whole point of the first SequentialScan is to get our pointer aligned.  That's what:

```
length = (((int)pCh & (Unsafe.SizeOf<Vector<ushort>>() - 1)) / elementsPerByte) + 1;
```

that line is doing.

Once `pCh` is aligned, `pStart` will be aligned because:

`char* pStart = pCh - Vector<ushort>.Count + 1;`